pilot/pkg/model/sidecar.go

```diff
@@ -143,18 +149,41 @@ func DefaultSidecarScopeForNamespace(ps *PushContext, configNamespace string) *S
 	meshGateway := map[string]bool{constants.IstioMeshGateway: true}
 	defaultEgressListener.virtualServices = ps.VirtualServices(&dummyNode, meshGateway)
 
+	serviceDependencies := make(map[string]struct{}, len(defaultEgressListener.services))
+	virtualServiceDependencies := make(map[string]struct{})
+	destinationRuleDependencies := make(map[string]struct{})
+
```

Comment:
Make this also configDependencies please, else this will keep growing.

Reply:
done